indexing-hadoop/src/main/java/io/druid/indexer/InputRowHelper.java

```diff
@@ -0,0 +1,259 @@
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.indexer;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.Maps;
+import com.google.common.io.ByteArrayDataOutput;
+import com.google.common.io.ByteStreams;
+import com.metamx.common.ISE;
+import com.metamx.common.logger.Logger;
+import io.druid.data.input.InputRow;
+import io.druid.data.input.MapBasedInputRow;
+import io.druid.data.input.Row;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.segment.incremental.IncrementalIndex;
+import io.druid.segment.serde.ComplexMetricSerde;
+import io.druid.segment.serde.ComplexMetrics;
+import org.apache.hadoop.io.ArrayWritable;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.MapWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.UTF8;
+import org.apache.hadoop.io.Writable;
+import org.joda.time.DateTime;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ */
+public class InputRowHelper
+{
+  private static final Logger log = new Logger(InputRowHelper.class);
+
+  private static final String[] EMPTY_STR_ARRAY = new String[0];
+
+  public final static byte[] toBytes(InputRow row, AggregatorFactory[] aggs)
+  {
+    try {
+      ByteArrayDataOutput out = ByteStreams.newDataOutput();
+
+      List<String> dimList = row.getDimensions();
+      String[] dims = dimList == null ? EMPTY_STR_ARRAY : dimList.toArray(EMPTY_STR_ARRAY);
+      StringArrayWritable sw = new StringArrayWritable(dims);
+      sw.write(out);
+
+      out.writeLong(row.getTimestampFromEpoch());
+
+      MapWritable mw = new MapWritable();
+
+      //writing all dimensions
+      for (String dim : dims) {
+        List<String> dimValue = row.getDimension(dim);
+
+        if (dimValue == null || dimValue.size() == 0) {
+          continue;
+        }
+
+        if (dimValue.size() == 1) {
+          mw.put(new Text(dim), new Text(dimValue.get(0)));
+        } else {
+          mw.put(new Text(dim), new StringArrayWritable(dimValue.toArray(EMPTY_STR_ARRAY)));
+        }
+      }
+
+      //writing all metrics
+      for (AggregatorFactory aggFactory : aggs) {
+        String k = aggFactory.getName();
+
+        ThreadLocal<InputRow> trow = new ThreadLocal<>();
+        trow.set(row);
+        Aggregator agg = aggFactory.factorize(
+            IncrementalIndex.makeColumnSelectorFactory(
```

Comment:
This is a bit of a different usage pattern compared to many other places. Typically setup like column selector factory occurs at a more general level, and the data is simply run through the data. As such it would fit in better here to have a way to have the factories exist on a larger scope (maybe calling method?) and have factorize simply call them.